### PR TITLE
phase out julia 0.6

### PR DESCRIPTION
**Pull request by dpo**

### CODE REVIEW

**Comment by lrsantos11** [comment]
Look at PR #20

**Comment by lrsantos11** [comment]
@dpo I cannot help you because I cannot write on HSL.jl repo.

File: .travis.yml [hunk 1]
```diff
@@ -5,19 +5,18 @@ os:
...
   email: false
-# uncomment the following lines to override the default test script
-#script:
-#  - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
-#  - julia --check-bounds=yes -e 'Pkg.clone(pwd()); Pkg.build("MA97"); Pkg.test("MA97"; coverage=true)'
+
+script:
+ - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
+ - julia --check-bounds=yes -e 'using Pkg; Pkg.clone(pwd()); Pkg.build("HSL"); Pkg.test("HSL"; coverage=true)'
+ - julia --check-bounds=yes -e 'using Pkg; Pkg.add("BinDeps"); Pkg.add("Compat"); Pkg.build("BinDeps"); using BinDeps; BinDeps.debug("HSL")'
```
Comment (lrsantos11) [inline .travis.yml]: The problem is that `BinDeps` is not working correctly on Julia 1.0. I suggest to comment such dependency test for when this Pkg is working again. 
```suggestion
  # - julia --check-bounds=yes -e 'using Pkg; Pkg.add("BinDeps"); Pkg.add("Compat"); Pkg.build("BinDeps"); using BinDeps; BinDeps.debug("HSL")'
```

**Comment by lrsantos11** [comment]
As far as I can see, now for Julia 1.0+ one should use https://github.com/JuliaPackaging/BinaryBuilder.jl and https://github.com/JuliaPackaging/BinaryProvider.jl instead of BinDeps.jl.

**Comment by lrsantos11** [comment]
@dpo According to https://github.com/JuliaPackaging/BinaryBuilder.jl README, https://github.com/JuliaLinearAlgebra/OpenBLASBuilder is just an automatic Builder that was made using BinaryBuilder. I guess we have to do it by ourselves.

**Comment by lrsantos11** [comment]
@dpo Here  (https://github.com/JuliaPackaging/BinaryProvider.jl/tree/master/test/LibFoo.jl) you can find the example that you asked for. Since HSL library cannot be available online due to license restrictions, I don't know how we can manage to use it yet. I'm studying BinaryPackage.jl manual to figure out.

**Comment by dpo** [comment]
I tried including the build file for OpenBLAS provided by OpenBLASBuilder (https://github.com/JuliaLinearAlgebra/OpenBLASBuilder/releases/download/v0.3.0-3/build_OpenBLAS.v0.3.0.jl) from `build_blas.jl` with no success. Could you let me know if it works for you?

**Comment by dpo** [comment]
Does including the OpenBLAS build script instead of `build_blas.jl` work for you on linux?

**Comment by abelsiqueira** [comment]
It didn't. I tried some changes to make BinaryProvider and BinDeps talk, but it didn't work. I think dropping BinDeps is better.

**Comment by abelsiqueira** [comment]
@lrsantos11 we might be able to use [CoinMETIS](https://github.com/JuliaOpt/COINMetisBuilder/blob/master/build_tarballs.jl).

**Comment by lrsantos11** [comment]
> * A new possibility to try to simplify the build process is
>   
>   * previously building the dependencies with `BinaryBuilder.jl`, storing them somewhere,
>   * and downloading it when needed with `BinaryProvider.jl`;
>   * e.g. [`Ipopt`](https://github.com/JuliaOpt/Ipopt.jl/blob/master/deps/build.jl)
Agree with @abelsiqueira. Drop BinDeps and use `BinaryBuilder.jl`. 

> * The new possibility is entirely optional, and we could add a long list of if-else and `make` in our `build` script (though I'm not sure how `deps.jl` would be generated).
AFIK, `BinaryBuilder.jl` generates it automatically.

> Can't use CoinMETIS at the moment because it doesn't produce an .so ([JuliaOpt/COINMetisBuilder#1](https://github.com/JuliaOpt/COINMetisBuilder/issues/1)). Maybe modifying it is better than creating another Provider.

Will @juan-pablo-vielma be ok with it? 

One enhancement that I would like to have on `HSL.jl` is
*  both MA57 and MA97 working.

**Comment by dpo** [comment]
You can build an `so` by first compiling with `-fPIC` and then using this strategy: https://github.com/JuliaSmoothOptimizers/ThinASLBuilder/blob/master/build_tarballs.jl#L55

**Comment by abelsiqueira** [comment]
Tried creating a `.so` using COINMetis but failed.
I'm getting after a `.la` after `make` and the strategy above of converting it to `.so` isn't working. I can do it with direct download of Metis, so I'll look into that.

**Comment by abelsiqueira** [comment]
I created a [MetisBuilder](https://github.com/JuliaSmoothOptimizers/MetisBuilder), though I still need to configure it to automatically build the tarballs.

**Comment by dpo** [comment]
FINALLY!!!

**Pull request merged**